public/app/core/AppWrapper.tsx

```diff
@@ -0,0 +1,161 @@
+import angular from 'angular';
+import { each, extend } from 'lodash';
+import { config, getLocationService } from '@grafana/runtime';
+import { Provider } from 'react-redux';
+import { store } from 'app/store/store';
+import { ErrorBoundaryAlert, ModalRoot, ModalsProvider } from '@grafana/ui';
+import { GrafanaApp } from '../app';
+import { routes } from 'app/routes/routes';
+import { ConfigContext, ThemeProvider } from './utils/ConfigProvider';
+import { RouteDescriptor } from './navigation/types';
+import { contextSrv } from './services/context_srv';
+import { SideMenu } from './components/sidemenu/SideMenu';
+import { navigationLogger, queryStringToJSON, shouldForceReload } from './navigation/utils';
+import { updateLocation } from './actions';
+import { GrafanaRoute } from './navigation/GrafanaRoute';
+
+interface AppWrapperProps {
+  app: GrafanaApp;
+}
+
+interface AppWrapperState {
+  ngInjector: any;
+}
+
+export class AppWrapper extends React.Component<AppWrapperProps, AppWrapperState> {
+  container = React.createRef<HTMLDivElement>();
+
+  constructor(props: AppWrapperProps) {
+    super(props);
+
+    this.state = {
+      ngInjector: null,
+    };
+  }
+
+  componentDidMount() {
+    if (this.container) {
+      this.bootstrapNgApp();
+    } else {
+      throw new Error('Failed to boot angular app, no container to attach to');
+    }
+  }
+
+  bootstrapNgApp() {
+    const { app } = this.props;
+    const invoker = angular.bootstrap(document, app.ngModuleDependencies);
+    navigationLogger('AppWrapper', false, 'Angular app bootstrap');
+    this.setState(
+      { ngInjector: invoker },
+      invoker.invoke(() => {
+        each(app.preBootModules, (module) => {
+          extend(module, app.registerFunctions);
+        });
+        app.preBootModules = null;
+        // I don't know
+        return () => {};
+      })
+    );
+  }
+
+  renderRoute = (route: RouteDescriptor) => {
+    // const { updateLocation } = this.props;
+    // TODO[Router]
+    // @ts-ignore
+    const isAngularRoute = !!route.controller;
+    // TODO[Router]
+    // @ts-ignore
+    const roles = route.roles ? route.roles() : [];
+
+    if (isAngularRoute) {
+      return null;
+    }
+
+    return (
+      <Route
+        exact
+        path={route.path}
+        key={`${route.path}`}
+        render={(props) => {
+          navigationLogger('AppWrapper', false, 'Rendering route', route, 'with match', props.location);
+
+          if (shouldForceReload(props.location.search)) {
+            navigationLogger('AppWrapper', false, 'Force login', props.location);
+            window.location.href = `${props.location.pathname}${props.location.search}`;
+            return null;
+          }
+
+          store.dispatch(
+            updateLocation({
+              path: props.location.pathname,
+              routeParams: props.match.params,
+              query: queryStringToJSON(props.location.search),
+            })
+          );
```

Comment:
Calling a redux action during rendering seems bad. Is there a callback on Route that we can hook into instead?

Reply:
Indeed, that's gonna land in the `GrafanaRoute` component lifecycle.

Reply:
@torkelo thought about this a bit and my conclusion is to keep it in the render cycle. I've moved it to `SyncLocationWithRedux` intermediate component to make sure the dispatch happens before the component is rendered. I know it feels a weird pattern but the render is predictable here as it should happen only on route changes/matches. See the test and implementatin in 5eb4c5f.

Alternatively, we could hack something around the history.listen and use react-routers matchPath (https://github.com/ReactTraining/react-router/blob/master/packages/react-router/modules/Route.js#L38). But again the problem is we don't have the path to match the location against in that listener.

Reply:
ok, looks good. As long as we do not get warnings that we are updating state during rendering, but should not as this is redux state not react component state

Reply:
The more I explore this the more I think I'll just remove the location from redux altogether. The above introduces loads of react warnings(same as the previous approach) and looking at flips implemented in https://github.com/supasate/connected-react-router it only makes me more confident this complexity does not bring us much.

Reply:
flips?

Reply:
Flips, tricks, you name it:)